src/services/entitlement.service.ts

```diff
@@ -1,32 +1,36 @@
 import InPlayer from '@inplayer-org/inplayer.js';
+import { injectable } from 'inversify';
 
 import type { GetMediaParams } from '#types/media';
 
-const getToken = async <T>(url: string, body: unknown = {}, jwt?: string): Promise<T> => {
-  const response = await fetch(url, {
-    method: 'POST',
-    headers: {
-      Authorization: jwt ? `Bearer ${jwt}` : '',
-    },
-    body: JSON.stringify(body),
-  });
+@injectable()
+export default class EntitlementService {
```

Comment:
Probably we can split this into 2 services: `GenericEntitlementService` and `JwpEntitlementService`